src/common/history.c

```diff
@@ -1114,7 +1125,9 @@ static void _history_reorder(int32_t imgid)
     if (give_reorder_information) fprintf(stderr,", reorder\n");
     // make sure running jobs can't interfere here as the followiing code uses a fixed dummy id
     // and also intends to have a "properly" orderered database
-    dt_pthread_mutex_lock(&darktable.db_insert);
+    dt_pthread_mutex_lock(&darktable.db_multi_image);
+    dt_database_lock_image(imgid);
+    dt_database_lock_image(dummy);
```

Comment:
Likewise, dangerous.

Reply:
dummy doesn't have its own lock, it shares it with 1/64 of all images.